primitives/utility/src/lib.rs

```diff
@@ -51,3 +61,191 @@ impl<T: UpwardMessageSender, W: WrapVersion> SendXcm for ParentAsUmp<T, W> {
 		}
 	}
 }
+
+/// Contains information to handle refund/payment for xcm-execution
+#[derive(Clone, Eq, PartialEq, Debug)]
+struct AssetTraderRefunder<Balance, AssetId> {
+	// The amount of weight bought minus the weigh already refunded
+	weight: Weight,
+	// The balance paid for the weight bought minus the balance already refunded
+	balance: Balance,
+	// The multilocation representing the fungible asset in which fees were paid
+	location: MultiLocation,
```

Comment:
This is assuming that the asset is `Concrete`, which might not be the case. I suppose this should really be an `AssetId` instead.